shell/platform/darwin/ios/ios_external_texture_gl.mm

```diff
@@ -57,9 +60,9 @@
   GrGLTextureInfo textureInfo = {CVOpenGLESTextureGetTarget(texture_ref_),
                                  CVOpenGLESTextureGetName(texture_ref_), GL_RGBA8_OES};
   GrBackendTexture backendTexture(bounds.width(), bounds.height(), GrMipMapped::kNo, textureInfo);
-  sk_sp<SkImage> image =
-      SkImage::MakeFromTexture(canvas.getGrContext(), backendTexture, kTopLeft_GrSurfaceOrigin,
-                               kRGBA_8888_SkColorType, kPremul_SkAlphaType, nullptr);
+  sk_sp<SkImage> image = SkImage::MakeFromTexture(
+      context == nullptr ? canvas.getGrContext() : context, backendTexture,
```

Comment:
I think we can use the `context` safely. If `context` is `nullptr`, I don't see how `canvas.getGrContext` can be non-null.

Reply:
Why is the `canvas` getting the wrong GrContext?

Reply:
It's a recording canvas, it doesn't have any GrContext associated with it. It's null. When we upload a texture from a plugin on iOS, we upload it with the EAGLContext, when we try to make an `SKImage` using the recording canvas' GrContext, it fails.

Which reminds me, @Qxyat , as a part of this fix, could you add an error in `ios_external_texture_gl.mm`, like so:

```
  if (image) {
    canvas.drawImage(image, bounds.x(), bounds.y());
  } else {
    FML_LOG(ERROR) << "Unable to create an SKImage from Texture.";
  }
```

Reply:
Ok.  How about just

```
FML_CHECK(image) << "Failed to create SkImage from Texture.";
canvas.drawImage(image, bounds.x(), bounds.y());
```